addons/addon-base-raas/packages/base-raas-services/lib/environment/service-catalog/__tests__/environment-sc-service.test.js

```diff
@@ -1422,4 +1422,112 @@ describe('EnvironmentSCService', () => {
     expect(currentIngressRules).toMatchObject(expectedOutcome);
     expect(securityGroupId).toBeUndefined();
   });
+
+  describe('pollAndSyncSageMakerStatus', () => {
+    const roleArn = 'roleArn';
+    const externalId = 'externalId';
+    const requestContext = {};
+
+    const empty = {};
+
+    it('should finish updating before returning', async () => {
+      // BUILD
+      const sagemakerInstances = { 'notebook-instance-name': {} };
+      service.pollSageMakerRealtimeStatus = jest.fn(() => {
+        return { 'notebook-instance-name': 'InService' };
+      });
+      service.updateStatus = jest.fn(async () => {
+        // sleep
+        await new Promise(r => setTimeout(r, 4000));
```

Comment:
Can we make this `1000` and the original code without the bug fix will still fail this unit test? If yes, let's shorten the time frame so the test runs faster.

Reply:
let me see. If so, I will make it shorter.